pkg/deployment/reconcile/plan_executor.go

```diff
@@ -96,14 +98,15 @@ func (d *Reconciler) ExecutePlan(ctx context.Context) (bool, error) {
 				return false, maskAny(err)
 			}
 			if ready {
-				// action.CheckProgress may have changed status, so reload it.
-				status = d.context.GetStatus()
-				// Remove action from list
-				status.Plan = status.Plan[1:]
-				// Save plan update
-				if err := d.context.UpdateStatus(status); err != nil {
-					log.Debug().Err(err).Msg("Failed to update CR status")
-					return false, maskAny(err)
+				{ // action.CheckProgress may have changed status, so reload it.
```

Comment:
What does this extra pair of braces achieve?

Reply:
It provides an scope in which we're sure we're using the right `status`.